projects/hslayers/src/components/layermanager/layermanager-metadata.service.ts

```diff
@@ -157,39 +171,44 @@ export class HsLayerManagerMetadataService {
         const layerSubObject = this.identifyLayerObject(subLayer, layerCaps);
         layerObjs.push(layerSubObject);
         this.collectLegend(layerSubObject, legends);
-        if (layerSubObject.Layer !== undefined && getSubLayers(layer)) {
+        if (
+          layerSubObject &&
+          layerSubObject.Layer !== undefined &&
+          getSubLayers(olLayer)
+        ) {
           delete layerSubObject.Layer;
         }
       }
-      if (getCachedCapabilities(layer) == undefined) {
+      if (getCachedCapabilities(olLayer) === undefined) {
         layerObj = Object.assign(JSON.parse(JSON.stringify(layerObjs[0])), {
           maxResolution: Math.max(
             ...layerObjs.map((layer) => this.searchForScaleDenominator(layer))
           ),
           Layer: layerObjs,
         });
       }
-      this.fillMetadataUrlsIfNotExist(layer, caps);
+      this.fillMetadataUrlsIfNotExist(olLayer, caps);
     } else {
       layerObj = this.identifyLayerObject(layerName, layerCaps);
-      if (layerObj.Layer && getSubLayers(layer)) {
+      console.log('parsing..', layerObj);
+      if (layerObj.Layer && getSubLayers(olLayer) && fromSublayerParam) {
```

Comment:
If getSubLayers(olLayer) != undefined , then we know its from sublayes and dont need extra parameter fromSublayerParam

Reply:
Yes, it seems so. Thanks!